incubator/group/keeper_test.go

```diff
@@ -343,6 +343,258 @@ func TestCreateProposal(t *testing.T) {
+			srcVoters:     []sdk.AccAddress{[]byte("valid-member-address")},
+			srcChoice:     group.Choice_NO,
+			expErr:        true,
+		},
+		"existing proposal required": {
+			srcProposalID: 9999,
+			srcVoters:     []sdk.AccAddress{[]byte("valid-member-address")},
+			srcChoice:     group.Choice_NO,
+			expErr:        true,
+		},
+		"empty choice": {
+			srcProposalID: myProposalID,
+			srcVoters:     []sdk.AccAddress{[]byte("valid-member-address")},
+			expErr:        true,
+		},
+		"invalid choice": {
+			srcProposalID: myProposalID,
+			srcVoters:     []sdk.AccAddress{[]byte("valid-member-address")},
+			srcChoice:     5,
+			expErr:        true,
+		},
+		"all voters must be in group": {
+			srcProposalID: myProposalID,
+			srcVoters:     []sdk.AccAddress{[]byte("valid-member-address"), []byte("non--member-address")},
+			srcChoice:     group.Choice_NO,
+			expErr:        true,
+		},
+		"voters must not include nil": {
+			srcProposalID: myProposalID,
+			srcVoters:     []sdk.AccAddress{[]byte("valid-member-address"), nil},
+			srcChoice:     group.Choice_NO,
+			expErr:        true,
+		},
+		"voters must not be nil": {
+			srcProposalID: myProposalID,
+			srcChoice:     group.Choice_NO,
+			expErr:        true,
+		},
+		"voters must not be empty": {
+			srcProposalID: myProposalID,
+			srcChoice:     group.Choice_NO,
+			srcVoters:     []sdk.AccAddress{},
+			expErr:        true,
+		},
+		"admin that is not a group member can not vote": {
+			srcProposalID: myProposalID,
+			srcVoters:     []sdk.AccAddress{[]byte("valid--admin-address")},
+			srcChoice:     group.Choice_NO,
+			expErr:        true,
+		},
+		"after timeout": {
+			srcProposalID: myProposalID,
+			srcVoters:     []sdk.AccAddress{[]byte("valid-member-address")},
+			srcChoice:     group.Choice_NO,
+			srcCtx:        parentCtx.WithBlockTime(blockTime.Add(time.Second)),
+			expErr:        true,
+		},
+		"closed already": {
+			srcProposalID: myProposalID,
+			srcVoters:     []sdk.AccAddress{[]byte("valid-member-address")},
+			srcChoice:     group.Choice_NO,
+			doBefore: func(ctx sdk.Context) {
+				err := k.Vote(ctx, myProposalID, []sdk.AccAddress{[]byte("power-member-address")}, group.Choice_YES, "")
+				require.NoError(t, err)
+			},
+			expErr: true,
+		},
+		"voted already": {
+			srcProposalID: myProposalID,
+			srcVoters:     []sdk.AccAddress{[]byte("valid-member-address")},
+			srcChoice:     group.Choice_NO,
+			doBefore: func(ctx sdk.Context) {
+				err := k.Vote(ctx, myProposalID, []sdk.AccAddress{[]byte("valid-member-address")}, group.Choice_YES, "")
+				require.NoError(t, err)
+			},
+			expErr: true,
+		},
+		"with group modified": {
+			srcProposalID: myProposalID,
+			srcVoters:     []sdk.AccAddress{[]byte("valid-member-address")},
+			srcChoice:     group.Choice_NO,
+			doBefore: func(ctx sdk.Context) {
+				g, err := k.GetGroup(ctx, myGroupID)
+				require.NoError(t, err)
+				g.Comment = "modifed"
+				require.NoError(t, k.UpdateGroup(ctx, &g))
```

Comment:
For me this is an optimization. Simplest thing was to have any modification to upgrade the group version. Let's have a ticket to address this later. Another approach may be to keep historical versions so that modifications do not matter.

Reply:
Can you create another ticket to track this? I would prefer `Comment` and `Admin` don't invalidate proposals.

I think historical versions are more challenging with this ORM approach which allows for large group sizes.

Reply:
Opened https://github.com/regen-network/cosmos-modules/issues/65